lib/media/drm_engine.js

```diff
@@ -387,7 +387,50 @@ shaka.media.DrmEngine = class {
   }
 
   /**
-   * Attach MediaKeys to the video element and start processing events.
+   * Attach MediaKeys to the video element
+   * @return {!Promise}
+   * @private
+   */
+  async attachMediaKeys_() {
+    if (this.video_.mediaKeys) {
+      return;
+    }
+
+    try {
+      await this.video_.setMediaKeys(this.mediaKeys_);
+    } catch (exception) {
+      goog.asserts.assert(exception instanceof Error, 'Wrong error type!');
+
+      this.onError_(new shaka.util.Error(
+          shaka.util.Error.Severity.CRITICAL,
+          shaka.util.Error.Category.DRM,
+          shaka.util.Error.Code.FAILED_TO_ATTACH_TO_VIDEO,
+          exception.message));
+    }
+
+    this.destroyer_.ensureNotDestroyed();
+  }
+
+  /**
+   * Processes encrypted event and start licence challenging
+   * @return {!Promise}
+   * @private
+   */
+  async onEncryptedEvent_(event) {
+    /**
+     * MediaKeys should be added when receiving an encrypted event. Setting
+     * mediaKeys before could result into encrypted event not being fired on
+     * some browsers
+     */
+    await this.attachMediaKeys_();
+
+    this.newInitData(
+        event.initDataType,
+        shaka.util.BufferUtils.toUint8(event.initData));
+  }
+
+  /**
+   * Start processing events.
```

Comment:
The old comment for this method is still accurate, right? This still will attach media keys to the video, just not necessarily immediately, anymore.

Reply:
It is still accurate :)